analysis/src/code.rs

```diff
@@ -254,6 +294,32 @@ impl CodeAnalyzer {
         let geiger_report: GeigerReport = serde_json::from_slice(&output.stdout)?;
         Ok(geiger_report)
     }
+
+    fn get_unsafe_report(&self, name: String, version: String) -> Result<Option<UnsafeReport>> {
+        let key = (name, version);
+        let cache = self.geiger_cache.borrow();
```

Comment:
Maybe there should be a `get_cargo_geiger_report` that grabs one from the cache based on the key? That would hide the cache a bit from the other code.

Reply:
Probably it would return `Option<CargoGeigerReport>`.